JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -17,54 +17,78 @@
 package com.example.jetnews.ui.home
 
 import androidx.compose.Composable
+import androidx.compose.remember
+import androidx.ui.core.Modifier
 import androidx.ui.core.Opacity
 import androidx.ui.core.Text
 import androidx.ui.foundation.Clickable
 import androidx.ui.foundation.HorizontalScroller
 import androidx.ui.foundation.VerticalScroller
 import androidx.ui.layout.Column
 import androidx.ui.layout.LayoutPadding
-import androidx.ui.layout.LayoutWidth
 import androidx.ui.layout.Row
-import androidx.ui.layout.Spacer
 import androidx.ui.material.Divider
+import androidx.ui.material.DrawerState
 import androidx.ui.material.EmphasisLevels
 import androidx.ui.material.MaterialTheme
 import androidx.ui.material.ProvideEmphasis
+import androidx.ui.material.Scaffold
+import androidx.ui.material.ScaffoldState
 import androidx.ui.material.TopAppBar
 import androidx.ui.material.ripple.Ripple
 import androidx.ui.tooling.preview.Preview
 import androidx.ui.unit.dp
 import com.example.jetnews.R
 import com.example.jetnews.data.posts
 import com.example.jetnews.model.Post
+import com.example.jetnews.ui.AppDrawer
 import com.example.jetnews.ui.Screen
 import com.example.jetnews.ui.VectorImageButton
 import com.example.jetnews.ui.navigateTo
 
 @Composable
-fun HomeScreen(openDrawer: () -> Unit) {
+fun HomeScreen(scaffoldState: ScaffoldState = remember { ScaffoldState() }) {
+    Column {
+        Scaffold(
+            scaffoldState = scaffoldState,
+            drawerContent = {
+                AppDrawer(
+                    currentScreen = Screen.Home,
+                    closeDrawer = { scaffoldState.drawerState = DrawerState.Closed }
+                )
+            },
+            topAppBar = {
+                TopAppBar(
+                    title = { Text(text = "Jetnews") },
+                    navigationIcon = {
+                        VectorImageButton(R.drawable.ic_jetnews_logo) {
+                            scaffoldState.drawerState = DrawerState.Opened
+                        }
+                    }
+                )
+            },
+            bodyContent = {
```

Comment:
bodyContent is supposed to return a modifier (not sure if it's dev05 or dev06). We should pass that modifier to HomeScreenBody

Reply:
Good catch!